mango/clients/arcgis.py

```diff
@@ -85,47 +86,108 @@ def get_geolocation(self, address: str) -> tuple:
         destinations=schema,
     )
     def get_origin_destination_matrix(
-        self, *, origins: list, destinations: list, travel_mode: dict = None, sleep_time=2
+        self,
+        *,
+        mode: str = "sync",
+        origins: list,
+        destinations: list,
+        travel_mode: dict = None,
+        sleep_time: int = 2,
     ) -> list:
         """
         Get the origin destination matrix for a list of origins and destinations.
 
+        :param str mode: the mode to get the matrix. It can be "sync" or "async".
         :param list origins: a list of dictionaries with the origin information. Keys needed: "name", "x", "y".
         :param list destinations: a list of dictionaries with the destination information.
         Keys needed: "Name", "x", "y".
-        :param dict travel_mode:
+        :param dict travel_mode: a dictionary with the configuration for the travel mode
+        :param int sleep_time: the time to wait for the response in asynchronous mode
         :return: list of dictionaries with the origin destination matrix with distance in meters and time in seconds
         :rtype: list
         :doc-author: baobab soluciones
         """
+        if mode is None:
+            mode = "sync"
+        elif mode != "sync" and mode != "async":
+            warnings.warn(f"The selected mode {mode} is not valid. Using sync mode.")
+            mode = "sync"
+
         if travel_mode is None:
             travel_mode = ARCGIS_CAR_TRAVEL_MODE
 
-        # TODO: based on the size of the matrix attack the direct API or the job API.
         if len(origins) > 1000 or len(destinations) > 1000:
             raise NotImplementedError(
                 "The size of the matrix is too big. Maximum number of origins or destinations is 1000."
             )
 
+        if (len(origins) > 10 or len(destinations) > 10) and mode == "sync":
+            warnings.warn(
+                "The size of the matrix is too big for sync mode. Changing to async mode"
+            )
+            mode = "async"
+
         origins = {
             "features": [
                 {
                     "geometry": {"x": el["x"], "y": el["y"]},
-                    "attributes": {"Name": el["name"]},
+                    "attributes": {
+                        "Name": el["name"],
+                        "ObjectID": pos + 11,
+                    },
                 }
-                for el in origins
+                for pos, el in enumerate(origins)
             ]
         }
         destinations = {
             "features": [
                 {
                     "geometry": {"x": el["x"], "y": el["y"]},
-                    "attributes": {"Name": el["name"]},
+                    "attributes": {
+                        "Name": el["name"],
+                        "ObjectID": pos + 101,
```

Comment:
since this fucntion can accept 1000 origins or destinations, I think you should put a +1000 or even +10000

Reply:
Done